src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriter.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// This class provides helper methods for writing bytes using <see cref="TdsStream"/>
+    /// It extends <see cref="TdsBufferAlloc"/> that manages allocations of bytes buffer for better memory management.
+    /// </summary>
+    internal class TdsWriter : TdsBufferAlloc
+    {
+        private readonly TdsStream _stream;
+
+        /// <summary>
+        /// Instantiate TdsWriter with a TdsWriteStream
+        /// </summary>
+        /// <param name="stream">A Tds Writer stream to work with</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public TdsWriter(TdsStream stream)
+        {
+            _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+        }
+
+        #region Public APIs
+
+        /// <summary>
+        /// Writes short value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="v">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteShortAsync(short v, bool isAsync, CancellationToken ct)
```

Comment:
can you make there paramter names a bit more descriptive

e.g. 
```suggestion
        public async ValueTask WriteShortAsync(short shortValue, bool isAsync, CancellationToken ct)
```

Reply:
This applies to all the methods.